docs/source/compression/pruning_transformer.rst

```diff
@@ -0,0 +1,69 @@
+Pruning Transformer with NNI
+============================
+
+Workable Pruning Process
+------------------------
+
+Here we show an effective transformer pruning process that NNI team has tried, and users can use NNI to discover better processes.
+The full code can be found `here <https://github.com/microsoft/nni/tree/v2.9/examples/model_compress/pruning/pruning_bert_glue.py>`_.
+
+In simple terms, the entire pruning process can be divided into the following steps:
+
+1. Finetune the pretrained model on the downstream task. From our experience,
+   the final effect of pruning on the finetuned model is better than pruning directly on the pretrained model.
+   At the same time, the finetuned model obtained in this step will also be used as the teacher model for the following distillation training.
+2. Pruning the attention layer at first. Here we apply block-sparse on attention layer weight,
+   and directly prune the head (condense the weight) if the head was fully masked.
+   If the head was partial masked, we will not prune it and recover its weight.
+3. Retrain the head-pruned model with distillation. Recover the model precision before pruning FFN layer.
+4. Pruning the FFN layer. Here we apply the output channels pruning on the 1st FFN layer,
+   and the 2nd FFN layer input channels will be pruned due to the pruning of 1st layer output channels.
+5. Retrain the final pruned model with distillation.
+
+Following this process, we can get a good compression effect on transformer.
+
+Combined with the full code, some of our recipes:
```

Comment:
Are these implementation notes, or several insights / experience?

Reply:
experience is a better word

`Combined with the full code, some of our recipes:`
->
`During the process of pruning transformer, we gained some of the following experience:`